static-instrumenter/agent-instrumenter/src/main/java/io/opentelemetry/contrib/staticinstrumenter/ArchiveEntry.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.staticinstrumenter;
+
+class ArchiveEntry {
+
+  private static final ArchiveEntry NOT_CLASS =
+      new ArchiveEntry("", /* isClass= */ false, /* shouldBeSkipped= */ true);
+
+  private final String className;
+  private final String classPath;
+  private final boolean isClass;
```

Comment:
Storing this as a field seems like overkill (if `isClass()` is called 10 times per entry, maybe, if 1-3 times per entry, not such a big deal)